container-chains/runtime-templates/simple/src/lib.rs

```diff
@@ -1125,6 +1129,37 @@ impl_runtime_apis! {
             SLOT_DURATION
         }
     }
+
+    impl xcm_fee_payment_runtime_api::XcmPaymentApi<Block> for Runtime {
+        fn query_acceptable_payment_assets(xcm_version: staging_xcm::Version) -> Result<Vec<VersionedAssetId>, XcmPaymentApiError> {
+            if !matches!(xcm_version, 3 | 4) {
+                return Err(XcmPaymentApiError::UnhandledXcmVersion);
+            }
+            Ok([VersionedAssetId::V4(xcm_config::SelfReserve::get().into())]
+                .into_iter()
+                .filter_map(|asset| asset.into_version(xcm_version).ok())
+                .collect())
+        }
+
+        fn query_weight_to_asset_fee(weight: Weight, asset: VersionedAssetId) -> Result<u128, XcmPaymentApiError> {
+            let local_asset = VersionedAssetId::V4(xcm_config::SelfReserve::get().into());
+            let asset = asset
+                .into_version(4)
+                .map_err(|_| XcmPaymentApiError::VersionedConversionFailed)?;
+
+            if asset != local_asset { return Err(XcmPaymentApiError::AssetNotFound); }
```

Comment:
here we need to check if an asset-id exists for such an assetId in ForeignAsset. If it exists, we need to retrieve the asset-rate with such an assetId (if there is one).